sys/auto_init/auto_init.c

```diff
@@ -49,6 +49,11 @@ void auto_init(void)
         extern void init_schedstatistics(void);
         init_schedstatistics();
     }
+    if (IS_USED(MODULE_SCHED_ROUND_ROBIN)) {
+        LOG_DEBUG("Auto init sched_round_robin.\n");
+        extern void sched_round_robin_init(void);
+        sched_round_robin_init();
+    }
```

Comment:
Have you checked that this is actually needed?

`kernel_init()` will create the main thread and call `cpu_switch_context_exit()`, which in turn will run the scheduler which schedules the main thread (in absence of any other except the idle thread). The scheduler run should call `sched_runq_callback()` - which seems to be all that is needed to launch round robin scheduling.

Reply:
afaik (asked in matrix) you aren't supposed to set*timers until the *timer is initialized also afaik threads are created before autoinit happens -> threads are created before *timer is initalised -> I need to guard that.

Reply:
But note that there is one callback to `sched_runq_callback` prior to the auto-initialization, which is not guarded.

Reply:
but it does nothing until the init ran (static _current_rr_priority is initial 0)
the mask disable for prio == 0 
if prio != 0 than 0 in_current_rr_priority is never == prio
and 0 is also no 0xff;
therefor even though i runs it does nothing until 0xff is written there in the init()